go/store/nbs/journal.go

```diff
@@ -171,8 +184,33 @@ func trueUpBackingManifest(ctx context.Context, root hash.Hash, backing *journal
 	return mc, nil
 }
 
+// IterateRoots iterates over the in-memory roots tracked by the ChunkJournal and passes the root and associated
+// timestamp to a callback function, |f|. If |f| returns an error, iteration is stopped and the error is returned.
+func (j *ChunkJournal) IterateRoots(f func(root string, timestamp time.Time) error) error {
+	j.mu.Lock()
+	roots := j.roots
+	rootTimestamps := j.rootTimestamps
```

Comment:
Might want to make a note about what this locking is actually protecting here (you're making a consistent copy of both slice vars that won't change as other threads call `append` on them)